src/entities/factories/Factory.ts

```diff
@@ -2,30 +2,54 @@ import { merge } from 'lodash';
 import { Entity } from '../Entity';
 import { Context } from '../../Context';
 
+/**
+ * Represents an entity
+ */
 export interface EntityClass<T, U> {
   new (params: T & U, context: Context): Entity<T>;
+
+  /**
+   * unserialize serialized entity information
+   */
   unserialize(uid: string): U;
+
+  /**
+   * generate a unique identifier for an entity
+   */
   generateId(identifiers: U): string;
 }
 
+/**
+ * Abstract representing a factory that will build up properties of an entity
```

Comment:
Factories are tasked with creating instances of their corresponding Entity and managing the internal cache for that Entity type. So for example, the Security Token Factory is tasked with fetching necessary data to instance a Security Token, as well as fetching/refreshing the internal Security Token cache